utils/custom.js

```diff
@@ -100,6 +100,28 @@ const convertLocationFromModel = (locationModel, isTO = false) => {
   return '';
 };
 
+const setChildCommentLv2 = (childCommentLv2 = [], karmaScores, mySignUserId, myAnonymousId) => {
+  let new_child_comment_lv2 = childCommentLv2;
+  return new_child_comment_lv2.map((child2) => {
+    const child2_user = karmaScores.find((user) => user.user_id === child2.user_id);
+    if (child2.data.anon_user_info_emoji_name) {
+      return {
+        ...child2,
+        user_id: null,
+        user: {},
+        target_feeds: [],
+        is_you: myAnonymousId === child2.user_id,
+        karmaScores: roundingKarmaScore(child2_user?.karma_score || 0)
+      };
+    }
+    return {
+      ...child2,
+      is_you: mySignUserId === child2.user_id,
+      karmaScores: roundingKarmaScore(child2_user?.karma_score || 0)
+    };
+  });
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `setChildCommentLv2` function seems to be well implemented. It correctly checks if the user is anonymous or not and sets the `is_you` and `karmaScores` attributes accordingly. However, it's important to note that this function mutates the input array `childCommentLv2`. This could lead to unexpected side effects if the original array is used elsewhere in the code after this function is called. To avoid this, you can create a copy of the array before mapping over it.

```diff
-  let new_child_comment_lv2 = childCommentLv2;
+  let new_child_comment_lv2 = [...childCommentLv2];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->